plugins/utils/firenvim.nix

```
@@ -0,0 +1,86 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  types = lib.nixvim.nixvimTypes;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "firenvim";
+  defaultPackage = pkgs.vimPlugins.firenvim;
+
+  maintainers = with lib.maintainers; [ GaetanLepage ];
+
+  settingsOptions = {
+    globalSettings = defaultNullOpts.mkAttrsOf' {
+      type = types.anything;
+      pluginDefault = { };
+      example = {
+        alt = "all";
+      };
+      description = "Default settings that apply to all URLs.";
+    };
+
+    localSettings = defaultNullOpts.mkAttrsOf' {
+      type = with types; attrsOf anything;
+      pluginDefault = { };
+      example = {
+        ".*" = {
+          cmdline = "neovim";
+          content = "text";
+          priority = 0;
+          selector = "textarea";
+          takeover = "always";
+        };
+        "https?://[^/]+\\.co\\.uk/" = {
+          takeover = "never";
+          priority = 1;
+        };
+      };
+      description = ''
+        Map Javascript patterns that match against the full URL to settings that are used for all
+        URLs matched by that pattern.
+        When multiple patterns match a URL, the pattern with the highest "priority" value is used.
+      '';
+    };
+  };
+
+  settingsExample = {
+    globalSettings.alt = "all";
+    localSettings = {
+      ".*" = {
+        cmdline = "neovim";
+        content = "text";
+        priority = 0;
+        selector = "textarea";
+        takeover = "always";
+      };
+      "https?://[^/]+\\.co\\.uk/" = {
+        takeover = "never";
+        priority = 1;
+      };
+    };
+  };
+
+  settingsDescription = ''
+    Options fed to the `vim.g.firenvim_config` table.
+  '';
+
+  callSetup = false;
+  extraConfig = cfg: {
+    warnings =
+      lib.optional
+        (
+          config.performance.combinePlugins.enable
+          && (!(lib.elem "firenvim" config.performance.combinePlugins.standalonePlugins))
```

```suggestion
          && !(lib.elem "firenvim" config.performance.combinePlugins.standalonePlugins)
```

This should be fine, right? _Probably_ more readable without `()` spam, though it is subjective.

---

Sure, let's try !